pkg/cmd/server/api/types.go

```diff
@@ -0,0 +1,141 @@
+package api
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+)
+
+type OpenShiftAllInOneConfig struct {
+	api.TypeMeta
+
+	OpenShiftMasterConfig OpenShiftMasterConfig
+	NodeConfig            NodeConfig
+}
+
+type NodeConfig struct {
+	api.TypeMeta
+
+	NodeID string
+
+	NodeHost string
+
+	KubernetesAddress string
+	KubernetesCA      FileOrData
+
+	// Whether to enable TLS serving
+	TLS bool
+	// ServingCert is used to serve securely
+	ServingCert CertInfo
+
+	// The directory that volumes will be stored under
+	VolumeDir string
+
+	// The image used as the Kubelet network namespace and volume container.
+	NetworkContainerImage string
+
+	// If true, the Kubelet will ignore errors from Docker
+	AllowDisabledDocker bool
+
+	// NodeCert is the client configuration used identify this kubelet to the master
+	NodeCert CertInfo
```

Comment:
I think we actually need all the user info from the kubeconfig file... in GCE, CA, cert, key, *and* username and password are required